sympy/sets/conditionset.py

```diff
@@ -196,57 +200,39 @@ def free_symbols(self):
+        except TypeError:
+            return Contains(other, self, evaluate=False)
 
     def as_relational(self, other):
         return And(Lambda(self.sym, self.condition)(
             other), self.base_set.contains(other))
 
     def _eval_subs(self, old, new):
-        if not isinstance(self.sym, Expr):
-            # Don't do anything with the equation set syntax;
-            # that should go away, eventually.
-            return self
+        from sympy.utilities.iterables import flatten
         sym, cond, base = self.args
-        if old == sym:
-            # we try to be as lenient as possible to allow
-            # the dummy symbol to be changed
-            base = base.subs(old, new)
-            if isinstance(new, Symbol):
-                # if the assumptions don't match, the cond
-                # might evaluate or change
-                if (new.assumptions0 == old.assumptions0 or
-                        len(new.assumptions0) == 1 and
-                        old.is_commutative == new.is_commutative):
-                    if base != self.base_set:
-                        # it will be aggravating to have the dummy
-                        # symbol change if you are trying to target
-                        # the base set so if the base set is changed
-                        # leave the dummy symbol alone -- a second
-                        # subs will be needed to change the dummy
-                        return self.func(sym, cond, base)
-                    else:
-                        return self.func(new, cond.subs(old, new), base)
-                raise ValueError(filldedent('''
-                    A dummy symbol can only be
-                    replaced with a symbol having the same
-                    assumptions or one having a single assumption
-                    having the same commutativity.
-                '''))
-            # don't target cond: it is there to tell how
-            # the base set should be filtered and if new is not in
-            # the base set then this substitution is ignored
-            return self.func(sym, cond, base)
-        else:
-            cond = self.condition.subs(old, new)
-            base = self.base_set.subs(old, new)
-            # The condition may have become true due to assumptions
-            # on 'sym'. In order for .subs() to be consistent with
-            # __new__ we *don't* check if 'sym' actually belongs to
-            # 'base'. In other words: assumptions are ignored.
-            return self.func(self.sym, cond, base)
+        insym = old in flatten([sym])
+        # prioritize changing a symbol in the base
+        newbase = base.subs(old, new)
+        if newbase != base:
+            if not insym:
+                # XXX should we ignore the subs if TypeError is
+                # raised and just return self? e.g. if cond is
+                # x < 1 then subs(x, I) raises TypeError
+                cond = cond.subs(old, new)
```

Comment:
I think it's better just to let the exception raise unless we can think of a good reason for wanting to do such a substitution.